experimental/devices/mfrc522/rf522.go

```diff
@@ -0,0 +1,763 @@
+// Copyright 2016 The Periph Authors. All rights reserved.
+// Use of this source code is governed under the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+// Package mfrc522 controls a Mifare RFID card reader.
+// The code is largely ported from Python library : https://github.com/mxgxw/MFRC522-python
+// The datasheet is available at : https://www.nxp.com/docs/en/data-sheet/MFRC522.pdf
+package mfrc522
+
+import (
+	"errors"
+	"fmt"
+	"periph.io/x/periph/conn/gpio"
+	"periph.io/x/periph/conn/gpio/gpioreg"
+	"periph.io/x/periph/conn/spi"
+	"periph.io/x/periph/experimental/devices/mfrc522/commands"
+	"time"
+)
+
+// Dev defines the RFID reader properties, the SPI bus and GPIO pins.
+type Dev struct {
+	ResetPin      gpio.PinIO
+	IrqPin        gpio.PinIO
+	Authenticated bool
+	antennaGain   int
+	MaxSpeedHz    int
+	spiDev        spi.Conn
+}
+
+// DefaultKey provides the default bytes for card authentication for method B.
+var DefaultKey = []byte{0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF}
+
+// NewSPI creates and initializes the RFID card reader attached to SPI.
+// - spiPort - the SPI device to use.
+// - maxSpeed - maximum speed for SPI data transfers.
+// - resetPin - reset GPIO pin.
+// - irqPin - irq GPIO pin.
+func NewSPI(spiPort spi.Port, maxSpeed int, resetPin, irqPin string) (device *Dev, err error) {
+
+	spiDev, err := spiPort.Connect(int64(maxSpeed), spi.Mode0, 8)
+	if err != nil {
+		return
+	}
+
+	if err != nil {
+		return
+	}
+
+	dev := &Dev{
+		spiDev:     spiDev,
+		MaxSpeedHz: maxSpeed,
+	}
+
+	pin := gpioreg.ByName(resetPin)
+	dev.ResetPin = pin
+	dev.ResetPin.Out(gpio.High)
+
+	pin = gpioreg.ByName(irqPin)
+	dev.IrqPin = pin
+	dev.IrqPin.In(gpio.PullUp, gpio.FallingEdge)
+
+	err = dev.Init()
+
+	device = dev
+
+	return
+}
+
+// Init initializes the RFID chip.
+func (r *Dev) Init() (err error) {
```

Comment:
Same (about named arguments). Furthermore, there's a risk of aliasing which is easy to mess up.